include/aspect/material_model/interface.h

```diff
@@ -365,6 +398,11 @@ namespace aspect
        * in order to compute the reaction increment.
        */
       std::vector<std::vector<double> > reaction_terms;
+      
+      /**
+       * The derivatives used for forming the Jacobian, needed for the Newton solver.
+       */
+      MaterialModelDerivatives<dim> derivatives;
```

Comment:
@tjhei and @jdannberg : Do you think this is the right design to have a member variable in the `MaterialModelOutputs` variable, or would you prefer a `std_cxx11::shared_ptr<...>` pointer to a derivative object?

I figure that you probably want to have something similar in there for melt transport, and maybe it would be easier to have a collection of pointers to objects derived from a common base class, rather than storing a whole bunch of member variables here, all of which will rarely even be used.